### PR TITLE
Add support for new option 'basedir' so you can set where npm modules are resolved.

### PR DESCRIPTION
**Pull request by hswolff**
This adds a new option `basedir` so that you can configure from where a npm module is resolved from (using the [node-resolve API](https://github.com/substack/node-resolve)).

This is needed for my project where in my project I have a npm module (scaffold) which has its own npm module (theme). I use this plugin in my project to compile the scaffold code which uses the theme which uses normalize.css as a npm module. 

Basically my parent project is trying to compile my LESS but this plugin always looks for npm modules at the root when I want it to look in the context of a node_modules directory.

Sorry if that's confusing.

Let me know what I can do!

### CODE REVIEW

**Comment by gigstada** [comment]
Are there any plans to merge and release this? This would be a really great feature to have.

**Comment by hswolff** [comment]
👋 

Is this repo still alive?

**Comment by gigstada** [comment]
After some more digging on this, I discovered that I didn't actually need this feature because you can [add a "filename" property to the options object you pass into LESS.render()](http://lesscss.org/#using-less-configuration). This sets the location for resolving "npm://*" references. Effectively, you're saying, "This is is the filename where this LESS code came from." I'm not even sure the filename you give it has to exist. You may be able to anchor the resolving to a certain path just by doing "filename": "your/favorite/path/fake.less". If it does have to exist, I guess you could just put an empty file there.

Not sure that solves the all same use cases as this PR, but if it does, this can probably be closed. If someone REALLY needs this feature right now, I did fork and publish a [new version of this](https://www.npmjs.com/package/less-plugin-npm-import-reloaded) to npm (@hswolff may have done this too) before I discovered the method above. My fork will not be maintained, so I wouldn't recommend it.
